generators/chipyard/src/main/scala/ConfigFragments.scala

```diff
@@ -150,3 +150,12 @@ class WithControlCore extends Config((site, here, up) => {
     )
   case MaxHartIdBits => log2Up(up(RocketTilesKey, site).size + up(BoomTilesKey, site).size + 1)
 })
+
+
+/**
+ * Config fragment to use ChipTopCaughtReset as the top module, which adds a reset synchronizer to
+ * the top-level reset, allowing it to be asynchronous with the clock.
+ */
+class WithChipTopCaughtReset extends Config((site, here, up) => {
+  case BuildTop => (p: Parameters) => Module(new ChipTopCaughtReset()(p).suggestName("top"))
```

Comment:
Yeah, annotation is the right way to do it (https://github.com/ucb-bar/barstools/issues/77). Until we fix that I'll note in the docs that you have to change TOP.

Reply:
Noted

Reply:
I think its worth devising a way to avoid manual setting of TOP. Specifically, I think we should try to have just one default "ChipTop" class, similar to how we have just one default "DigitalTop" (System) class.

Reply:
OK, see comment above. For this PR we can do that, but I think we do want to make `TOP` programmatically generated anyway, which makes this less painful.